File: YEAR 2/SEM 1/Advanced Programming Methods/LAB 4/src/View/RunExampleCommand.java
package View;

import Controller.InterpreterController;

public class RunExampleCommand extends Command {
    private InterpreterController controller;

    public RunExampleCommand(String key,String description,InterpreterController controller) {
        super(key,description);
        this.controller = controller;
    }

    @Override
    public void execute() {
        try {
            this.controller.executeAllSteps();
        }catch(Exception ex){
            System.out.println(ex.getMessage());
        }
    }
}

File: YEAR 2/SEM 1/Advanced Programming Methods/ToyLanguage/ToyLanguage/View/RunExampleCommand.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using ToyLanguage.Controller;

namespace ToyLanguage.View
{
    class RunExampleCommand : Command
    {
        private readonly InterpreterController controller;
        public RunExampleCommand(string key, string description,InterpreterController controller) : base(key, description)
        {
            this.controller = controller;
        }

        public override void Execute()
        {
            try
            {
                controller.ExecuteAllSteps();
            }
            catch (Exception ex)
            {
                Console.WriteLine("An error ocurred: " + ex);
            }
        }
    }
}
